sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -574,6 +574,21 @@ case class DeleteFromTable(
   override def children: Seq[LogicalPlan] = child :: Nil
 }
 
+/**
+ * Update the data of table that specified with the condition with the given updated values.
+ * NOTE: Considering nested fields, we let the type of `attrs` to be `Seq[Expression]` becuase
+ * nested fields may be resolved to objects of type like `GetStructField`. However, currently
+ * we only support top-level fields, in which case `attrs` is actually of type `Seq[Attribute]`.
+ */
+case class UpdateTable(
+    child: LogicalPlan,
+    attrs: Seq[Expression],
```

Comment:
I think it would be better to continue using `Seq[Attribute]` until there is a better way to reference nested fields.

Reply:
I have a long discussing about this with @cloud-fan , see https://github.com/apache/spark/pull/25626#discussion_r320095937  and the relative comments. Let's go on the discussing there.

Reply:
Note that, since we don't forbid nested fields at parser level, analyzer can resolve the column name to either `Attribute` or nested field `GetStructField`. We fail the query at planning time if nested fields are found. Hopefully we can get rid of this limitation in the future.

If we change this to `Seq[Attribute]`, we must rely on type hack to hold `GetStructField` in `Seq[Attribute]`.